unit-testing/presentation.html

```diff
@@ -1,27 +1,310 @@
+            });​​
+            it('should test this', () => {​​
+             // ...​​
+            });​​
+          });
+          </code></pre>
+      </section>
+      <!-- 17 slide -->
+      <section>
+        <h3 class="slide-name">Expect​</h3>
+        <pre><code data-trim data-noescape>
+          expect(value)​​
+          expect.anything()​​
+          expect.any(constructor)​​
+          .not​​
+          .toBe(value)​​
+          .toHaveBeenCalled()​​
+          .toBeLessThan(number | bigint)​​
+          .toBeNull()​​
+          .toContain(item)​​
+          .toEqual(value)​​
+          .toMatch(regexp | string)​​
+          .toMatchSnapshot(propertyMatchers?, hint?)​​
+          .toThrow(error?)​​
+        </code></pre>
+      </section>
+      <!-- 18 slide -->
+      <section>
+        <h3 class="slide-name">Mocking & The Jest Object​</h3>
+        <pre><code data-trim data-noescape>
+          // Mock functions​​
+          jest.mock(moduleName, factory)​​
+          jest.unmock(moduleName)​​
+          jest.requireActual(moduleName)​​
+          jest.resetModules()​​
+          jest.isolateModules(fn)​​
+          // Mock modules​​
+          jest.fn(implementation?)​​
+          jest.spyOn(object, methodName)​​
+          jest.clearAllMocks()​​
+          jest.resetAllMocks()​​
+          jest.restoreAllMocks()​​
+        </code></pre>
+      </section>
+      <!-- 19 slide -->
+      <section>
+        <h3 class="slide-name">Snapshots​​</h3>
+        <pre><code data-trim data-noescape>
+          it('should match snapshot', () => {​​
+            const {toJSON} = render(<MyButton title="title" />);​​
+            expect(toJSON()).toMatchSnapshot();​​
+          });​​
+        </code></pre>
+      </section>
+      <!-- 20 slide -->
+      <section>
+        <!-- 20/0 slide -->
+        <section>
+          <h3 class="slide-name">Test coverage​</h3>
+          <p class="font-size-2 align-center">With that being said it is generally accepted that 80% coverage is a good
+            goal to aim for.</p>
+          <p class="font-size-2 align-center">Trying to reach a higher coverage might turn out to be costly, while not
+            necessary producing enough benefit.​</p>
+        </section>
+        <!-- 20/1 slide -->
+        <section>
+          <img src='assets/images/coverage_1.png' height="500"/>
+        </section>
+        <!-- 20/2 slide -->
+        <section>
+          <img src='assets/images/coverage_2.png' height="500"/>
+        </section>
+        <!-- 20/3 slide -->
+        <section>
+          <img src='assets/images/coverage_3.png' height="500"/>
+        </section>
+      </section>
+      <!-- 21 slide -->
+      <section>
+        <h3 class="slide-name">When tests fail​​</h3>
+        <img src='assets/images/failed_tests.png' height="500"/>
+      </section>
+      <!-- 22 slide -->
+      <section>
+        <h3 class="slide-name">Husky</h3>
+        <pre><code data-trim data-noescape>
```

Comment:
could you describe in few words what it is and benefits

Reply:
done